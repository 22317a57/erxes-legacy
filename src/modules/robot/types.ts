export type IFeature = {
  name: string;
  text: string;
};
