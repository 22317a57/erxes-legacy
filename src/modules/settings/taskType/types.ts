export interface ITaskType {
  _id: string;
  name: string;
  icon: string;
}

// query types

export type TaskTypeQueryResponse = {
  taskTypes: ITaskType[];
  loading: boolean;
  refetch: () => void;
};

export type MutationVariables = {
  _id?: string;
  name?: string;
  icon?: string;
};

// mutation types

export type AddMutationResponse = {
  addMutation: (mutation: { variables: MutationVariables }) => Promise<any>;
};

export type EditMutationResponse = {
  editMutation: (mutation: { variables: MutationVariables }) => Promise<any>;
};

export type RemoveMutationResponse = {
  removeMutation: (mutation: { variables: { _id: string } }) => Promise<any>;
};
