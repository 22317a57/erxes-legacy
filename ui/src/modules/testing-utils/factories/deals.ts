import * as Factory from 'factory.ts';
import { IBoard, IPipeline, IStage } from 'modules/boards/types';
import { IDeal } from 'modules/deals/types';
import { userFactory } from './auth';
import { companyFactory } from './companies';
import { customerFactory } from './customer/customer';

export const boardFactory = Factory.Sync.makeFactory<IBoard>({
  _id: '1',
  name: 'New Board'
});

const board = boardFactory.build();

export const pipelineFactory = Factory.Sync.makeFactory<IPipeline>({
  _id: '1',
  name: 'new Pipeline',
  boardId: board._id,
  visibility: 'public',
  isWatched: true
});

export const stageFactory = Factory.Sync.makeFactory<IStage>({
  _id: '1',
  name: 'New stage',
  type: 'deal',
  probability: 'probability',
  itemsTotalCount: 2,
  initialDealsTotalCount: 3,
  inProcessDealsTotalCount: 4,
  stayedDealsTotalCount: 5,
  compareNextStage: { count: 1, percent: 20 },
  formId: 'formId',
  pipelineId: 'pipelineId'
});

const stage = stageFactory.build();
const user = userFactory.build();
const company = companyFactory.build();
const customer = customerFactory.build();
const pipeline = pipelineFactory.build();

const dealItem = {
  _id: '1',
  name: 'New Item',
  order: 2,
  stageId: stage._id,
  boardId: board._id,
  closeDate: new Date(),
  description: 'string',
  amount: 300,
  modifiedAt: new Date(),
  assignedUsers: [user],
  companies: [company],
  customers: [customer],
  labels: [],
  pipeline,
  isComplete: false,
  reminderMinute: 20,
  labelIds: [],
  createdAt: new Date()
};

export const dealFactory = Factory.Sync.makeFactory<IDeal>(dealItem);
