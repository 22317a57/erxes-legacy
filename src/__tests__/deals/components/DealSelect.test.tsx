// test passed
import { mount, shallow } from 'enzyme';
import DealSelect from 'modules/deals/components/DealSelect';
import { IBoard, IPipeline, IStage } from 'modules/deals/types';
import * as React from 'react';
import * as renderer from 'react-test-renderer';

describe('DealSelect component', () => {
  const testBoards: IBoard[] = [
    {
      _id: 'qwe124',
      name: 'qwertu'
    },
    {
      _id: 'qwe123',
      name: 'qwerty'
    }
  ];

  const testPipeline: IPipeline[] = [
    {
      _id: 'q123',
      name: 'John'
    },
    {
      _id: 'asd123',
      name: 'Kate'
    }
  ];

  const testStage: IStage[] = [
    {
      _id: 'w123',
      dealsTotalCount: 231
    },
    {
      _id: 'q123',
      dealsTotalCount: 123
    }
  ];

  const defaultProps = {
    boards: testBoards,
    pipelines: testPipeline,
    stages: testStage,
    onChangeBoard: (value: string) => null,
    onChangePipeline: (value: string) => null,
    onChangeStage: (value: string, callback?: () => void) => null
  };

  test('renders shallow successfully', () => {
    shallow(<DealSelect {...defaultProps} />);
  });

  test('renders mount with default props', () => {
    const control = mount(<DealSelect {...defaultProps} />);
    const props = control.props();

    expect(props).toMatchObject(defaultProps);
  });

  test('snapshot matches', () => {
    const rendered = renderer.create(<DealSelect {...defaultProps} />).toJSON();

    expect(rendered).toMatchSnapshot();
  });
});
