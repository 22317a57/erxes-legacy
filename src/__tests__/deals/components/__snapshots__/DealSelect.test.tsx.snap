// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DealSelect component snapshot matches 1`] = `
<div
  className="sc-hwcHae hiyYjn"
>
  <div
    className="sc-jKVCRD eDSGVr"
  >
    <label
      className="sc-ebFjAB bHuvQk"
    >
      Board
    </label>
    <div
      className="Select is-searchable Select--single"
    >
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-5--value"
        >
          <div
            className="Select-placeholder"
          >
            Choose a board
          </div>
          <div
            className="Select-input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-activedescendant="react-select-5--value"
              aria-expanded="false"
              aria-haspopup="false"
              aria-owns=""
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              role="combobox"
              style={
                Object {
                  "boxSizing": "content-box",
                  "width": "5px",
                }
              }
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="sc-jKVCRD eDSGVr"
  >
    <label
      className="sc-ebFjAB bHuvQk"
    >
      Pipeline
    </label>
    <div
      className="Select is-searchable Select--single"
    >
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-6--value"
        >
          <div
            className="Select-placeholder"
          >
            Choose a pipeline
          </div>
          <div
            className="Select-input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-activedescendant="react-select-6--value"
              aria-expanded="false"
              aria-haspopup="false"
              aria-owns=""
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              role="combobox"
              style={
                Object {
                  "boxSizing": "content-box",
                  "width": "5px",
                }
              }
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
  </div>
  <div
    className="sc-jKVCRD eDSGVr"
  >
    <label
      className="sc-ebFjAB bHuvQk"
    >
      Stage
    </label>
    <div
      className="Select is-searchable Select--single"
    >
      <div
        className="Select-control"
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
      >
        <span
          className="Select-multi-value-wrapper"
          id="react-select-7--value"
        >
          <div
            className="Select-placeholder"
          >
            Choose a stage
          </div>
          <div
            className="Select-input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-activedescendant="react-select-7--value"
              aria-expanded="false"
              aria-haspopup="false"
              aria-owns=""
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={false}
              role="combobox"
              style={
                Object {
                  "boxSizing": "content-box",
                  "width": "5px",
                }
              }
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </span>
        <span
          className="Select-arrow-zone"
          onMouseDown={[Function]}
        >
          <span
            className="Select-arrow"
            onMouseDown={[Function]}
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
